From 200652955b9145b7e0cd5af819a3cd43981098c1 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 19 Jan 2017 15:05:29 +0800
Subject: [PATCH 1481/5242] MLK-13783 char: otp: no need to check bank0/bank1
 when prog

commit  81029c732a08b1c63ffeb28f3bc154573d7881c9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Bank0/Bank1 are not in ECC mode, so no need to check.
Each bank contains 8 words, so we check (phy_index > 15).

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/char/fsl_otp.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/char/fsl_otp.c b/drivers/char/fsl_otp.c
index 2c0aa4e..ebeb0fc 100644
--- a/drivers/char/fsl_otp.c
+++ b/drivers/char/fsl_otp.c
@@ -571,8 +571,8 @@ static ssize_t fsl_otp_store(struct kobject *kobj, struct kobj_attribute *attr,
 
 	mutex_lock(&otp_mutex);
 
-	if (fsl_otp->devtype == FSL_OTP_MX7ULP) {
-		phy_index = fsl_otp_word_physical(fsl_otp, index);
+	phy_index = fsl_otp_word_physical(fsl_otp, index);
+	if ((fsl_otp->devtype == FSL_OTP_MX7ULP) && (phy_index > 15)) {
 		fsl_otp->set_otp_timing();
 		ret = otp_wait_busy(0);
 		if (ret)
-- 
1.7.9.5

